src/app/dataset/reportsimple/reportsimple.component.ts

```diff
@@ -82,12 +82,18 @@ export class ReportSimpleComponent {
   */
   copyReport(): void {
     const element = this.contentRef.nativeElement;
-    window.getSelection().removeAllRanges();
+    const doIfSelectionPresent = (fn: (sel: Selection) => void): void => {
+      const selection = window.getSelection();
+      if (selection) {
+        fn(selection);
+      }
+    };
+    doIfSelectionPresent((sel: Selection) => sel.removeAllRanges());
     const range = document.createRange();
     range.selectNode(element);
-    window.getSelection().addRange(range);
```

Comment:
Isn't this the same as `window.getSelection()?.addRange(range);`?
Same for two lines down (line 96).

Reply:
Window.getSelection assumes there is a selection to be gotten.  The call to getSelection has been moved to an inline function.

Reply:
I'm not sure I understand your reply. :)

My point is that you don't need the inline function you created because all it does is check if there is a selection, and if there is, apply the operation to it. But we now have optional chaining for that (the `?` that is inserted in my suggestion). Wouldn't that apply here?